c-bindings/c-bindings.go

```diff
@@ -156,5 +160,236 @@ func jsonnet_tla_code(vmRef *C.struct_JsonnetVm, key, value *C.char) {
 	vm.TLACode(C.GoString(key), C.GoString(value))
 }
 
+//export jsonnet_native_callback
+func jsonnet_native_callback(vmRef *C.struct_JsonnetVm, name *C.char, cb *C.JsonnetNativeCallback, ctx unsafe.Pointer, params **C.char) {
+	vm := getVM(vmRef)
+	p := unsafe.Pointer(params)
+	sz := unsafe.Sizeof(*params)
+
+	var identifiers ast.Identifiers
+
+	for i := 0; ; i++ {
+		arg := (**C.char)(unsafe.Pointer(uintptr(p) + uintptr(i)*sz))
+
+		if *arg == nil {
+			break
+		}
+
+		identifiers = append(identifiers, ast.Identifier(C.GoString(*arg)))
+	}
+
+	f := &jsonnet.NativeFunction{
+		Name:   C.GoString(name),
+		Params: identifiers,
+		Func: func(x []interface{}) (interface{}, error) {
+			var (
+				arr     []*C.struct_JsonnetJsonValue
+				argv    **C.struct_JsonnetJsonValue
+				success = C.int(0)
+			)
+
+			if len(x) > 0 {
+				arr = make([]*C.struct_JsonnetJsonValue, 0, len(x))
+
+				for _, json := range x {
+					arr = append(arr, createJSONValue(vmRef, json))
+				}
+
+				argv = &(arr[0])
+			}
+
+			result := C.jsonnet_internal_execute_native(cb, ctx, argv, &success)
+			v := getJSONValue(result)
+
+			for _, val := range arr {
+				jsonnet_json_destroy(vmRef, val)
+			}
+
+			jsonnet_json_destroy(vmRef, result)
+
+			if success != 1 {
+				return nil, fmt.Errorf("failed to execute native callback, code: %d", success)
+			}
+
+			return v.obj, nil
+		},
+	}
+
+	vm.NativeFunction(f)
+}
+
+//export jsonnet_json_extract_string
+func jsonnet_json_extract_string(vmRef *C.struct_JsonnetVm, json *C.struct_JsonnetJsonValue) *C.char {
+	_ = getVM(vmRef) // Here we check it, is it ok?
```

Comment:
It is ok, but we don't have to validate it at all. C++ implementation also doesn't use vm for these functions (https://github.com/google/jsonnet/blob/master/core/libjsonnet.cpp#L55).